Modules/mmapmodule.c

```diff
@@ -936,6 +936,7 @@ mmap_subscript(mmap_object *self, PyObject *item)
     CHECK_VALID(NULL);
     if (PyIndex_Check(item)) {
         Py_ssize_t i = PyNumber_AsSsize_t(item, PyExc_IndexError);
+        CHECK_VALID(NULL);
```

Comment:
I think it's better to add it right before 
```c
        return PyLong_FromLong(Py_CHARMASK(self->data[i]));
```
which maintains the best compatibility.

Also we can add `CHECK_VALID(NULL);`